# Dota 2 Winner Prediction

## Data description

We have the following files:

- sample_submission.csv: example of a submission file

- train_matches.jsonl, test_matches.jsonl: full "raw" training data

- train_features.csv, test_features.csv: features created by organizers

- train_targets.csv: results of training games (including the winner)

## Features created by organizers

These are basic features which include simple players' statistics. Scroll to the end to see how to build these features from raw json files.

In [6]:
import os
import pandas as pd

PATH_TO_DATA = '../data/'

df_train_features = pd.read_csv(os.path.join(PATH_TO_DATA,
                                             'train_features.csv'),
                                    index_col='match_id_hash')
df_train_targets = pd.read_csv(os.path.join(PATH_TO_DATA,
                                            'train_targets.csv'),
                                   index_col='match_id_hash')

df_train_features.shape

(39675, 245)

We have ~ 40k games, each described by match_id_hash (game id) and 245 features. Also game_time is given - time (in secs) when the snapshot was taken.

duration_time - time (in secs) when the game was over.

In [3]:
df_train_features.head()

game_time  game_mode  lobby_type  \
match_id_hash                                                        
a400b8f29dece5f4d266f49f1ae2e98a        155         22           7   
b9c57c450ce74a2af79c9ce96fac144d        658          4           0   
6db558535151ea18ca70a6892197db41         21         23           0   
46a0ddce8f7ed2a8d9bd5edcbb925682        576         22           7   
b1b35ff97723d9b7ade1c9c3cf48f770        453         22           7   

                                  objectives_len  chat_len  r1_hero_id  \
match_id_hash                                                            
a400b8f29dece5f4d266f49f1ae2e98a               1        11          11   
b9c57c450ce74a2af79c9ce96fac144d               3        10          15   
6db558535151ea18ca70a6892197db41               0         0         101   
46a0ddce8f7ed2a8d9bd5edcbb925682               1         4          14   
b1b35ff97723d9b7ade1c9c3cf48f770               1         3          42   

                                  r1_kills  r1_deaths  r1_assists  r1_denies  \
match_id_hash                                                                  
a400b8f29dece5f4d266f49f1ae2e98a         0          0           0          0   
b9c57c450ce74a2af79c9ce96fac144d         7          2           0          7   
6db558535151ea18ca70a6892197db41         0          0           0          0   
46a0ddce8f7ed2a8d9bd5edcbb925682         1          0           3          1   
b1b35ff97723d9b7ade1c9c3cf48f770         0          1           1          0   

                                  ...  d5_stuns  d5_creeps_stacked  \
match_id_hash                     ...                                
a400b8f29dece5f4d266f49f1ae2e98a  ...  0.000000                  0   
b9c57c450ce74a2af79c9ce96fac144d  ...  0.000000                  0   
6db558535151ea18ca70a6892197db41  ...  0.000000                  0   
46a0ddce8f7ed2a8d9bd5edcbb925682  ...  8.664527                  3   
b1b35ff97723d9b7ade1c9c3cf48f770  ...  0.000000                  2   

                                  d5_camps_stacked  d5_rune_pickups  \
match_id_hash                                                         
a400b8f29dece5f4d266f49f1ae2e98a                 0                0   
b9c57c450ce74a2af79c9ce96fac144d                 0                0   
6db558535151ea18ca70a6892197db41                 0                0   
46a0ddce8f7ed2a8d9bd5edcbb925682                 1                3   
b1b35ff97723d9b7ade1c9c3cf48f770                 1                2   

                                  d5_firstblood_claimed  \
match_id_hash                                             
a400b8f29dece5f4d266f49f1ae2e98a                      0   
b9c57c450ce74a2af79c9ce96fac144d                      0   
6db558535151ea18ca70a6892197db41                      0   
46a0ddce8f7ed2a8d9bd5edcbb925682                      0   
b1b35ff97723d9b7ade1c9c3cf48f770                      0   

                                  d5_teamfight_participation  \
match_id_hash                                                  
a400b8f29dece5f4d266f49f1ae2e98a                        0.00   
b9c57c450ce74a2af79c9ce96fac144d                        0.00   
6db558535151ea18ca70a6892197db41                        0.00   
46a0ddce8f7ed2a8d9bd5edcbb925682                        0.00   
b1b35ff97723d9b7ade1c9c3cf48f770                        0.25   

                                  d5_towers_killed  d5_roshans_killed  \
match_id_hash                                                           
a400b8f29dece5f4d266f49f1ae2e98a                 0                  0   
b9c57c450ce74a2af79c9ce96fac144d                 0                  0   
6db558535151ea18ca70a6892197db41                 0                  0   
46a0ddce8f7ed2a8d9bd5edcbb925682                 0                  0   
b1b35ff97723d9b7ade1c9c3cf48f770                 0                  0   

                                  d5_obs_placed  d5_sen_placed  
match_id_hash                     

We are interested in the radiant_win column in train_targets.csv. All these features are not known during the game (they come "from future" as compared to game_time), so we have these features only for training data.

In [4]:
df_train_targets.head()

game_time  radiant_win  duration  \
match_id_hash                                                        
a400b8f29dece5f4d266f49f1ae2e98a        155        False       992   
b9c57c450ce74a2af79c9ce96fac144d        658         True      1154   
6db558535151ea18ca70a6892197db41         21         True      1503   
46a0ddce8f7ed2a8d9bd5edcbb925682        576         True      1952   
b1b35ff97723d9b7ade1c9c3cf48f770        453        False      2001   

                                  time_remaining next_roshan_team  
match_id_hash                                                      
a400b8f29dece5f4d266f49f1ae2e98a             837              NaN  
b9c57c450ce74a2af79c9ce96fac144d             496              NaN  
6db558535151ea18ca70a6892197db41            1482          Radiant  
46a0ddce8f7ed2a8d9bd5edcbb925682            1376              NaN  
b1b35ff97723d9b7ade1c9c3cf48f770            1548              NaN